app/assets/v2/js/pages/bounty_details.js

```diff
@@ -633,51 +633,181 @@ var attach_override_status = function() {
   });
 };
 
-
 var show_interest_modal = function() {
   var self = this;
+  let modals = $('#modalInterest');
+  let modalBody = $('#modalInterest .modal-content');
+  let modalUrl = `/interest/modal?redirect=${window.location.pathname}&pk=${document.result['pk']}`;
+
+  modals.on('show.bs.modal', function() {
+    modalBody.load(modalUrl, ()=> {
+      if (document.result['repo_type'] === 'private') {
+        $('#nda-upload').show();
+        $('#issueNDA').prop('required', true);
+        document.result.unsigned_nda ? $('.nda-download-link').attr('href', document.result.unsigned_nda.doc) : $('#nda-upload').hide();
+      }
 
-  setTimeout(function() {
-    var url = '/interest/modal?redirect=' + window.location.pathname + '&pk=' + document.result['pk'];
-
-    $.get(url, function(newHTML) {
-      var modal = $(newHTML).appendTo('body').modal({
-        modalClass: 'modal add-interest-modal'
-      });
-
-      var actionPlanForm = modal.find('form#action_plan');
-      var issueMessage = actionPlanForm.find('#issue_message');
+      let actionPlanForm = $('#action_plan');
+      let issueMessage = $('#issue_message');
 
       issueMessage.attr('placeholder', gettext('What steps will you take to complete this task? (min 30 chars)'));
 
-      modal.on('submit', function(event) {
+      actionPlanForm.on('submit', function(event) {
         event.preventDefault();
 
-        var msg = issueMessage.val().trim();
+        let msg = issueMessage.val().trim();
+        let issueNDA = $('#issueNDA')[0].files;
 
         if (!msg || msg.length < 30) {
           _alert({message: gettext('Please provide an action plan for this ticket. (min 30 chars)')}, 'error');
           return false;
         }
 
+        if (typeof issueNDA[0] !== 'undefined') {
+          const formData = new FormData();
+
+          formData.append('docs', issueNDA[0]);
+          formData.append('doc_type', 'signed_nda');
+          const ndaSend = {
+            url: '/api/v0.1/bountydocument',
+            method: 'POST',
+            data: formData,
+            processData: false,
+            dataType: 'json',
+            contentType: false
+          };
+
+          $.ajax(ndaSend).done(function(response) {
+            _alert(response.message, 'info');
+            add_interest(document.result['pk'], {
+              issue_message: msg,
+              signed_nda: response.bounty_doc_id
+            }).then(success => {
+              if (success) {
+                $(self).attr('href', '/uninterested');
+                $(self).find('span').text(gettext('Stop Work'));
+                $(self).parent().attr('title', '<div class="tooltip-info tooltip-sm">' + gettext('Notify the funder that you will not be working on this project') + '</div>');
+                $.modal.close();
+              }
+            }).catch((error) => {
+              if (error.responseJSON.error === 'You may only work on max of 3 issues at once.')
```

Comment:
Is there a better way we can check for this error? I worry that in the future we may change this error message and not realize that it will break this.